enclave/core/sgx/calls.c

```diff
@@ -817,7 +833,23 @@ oe_result_t oe_ocall(uint16_t func, uint64_t arg_in, uint64_t* arg_out)
             td->state = OE_TD_STATE_RUNNING;
         }
 
-        /* ORET here */
+        /* If FS was saved prior to the ocall, restore it. Assume that if FS
+         * was changed previously, then the wrfsbase instruction is available.
+         * Even if it isn't, we will emulate it in the exception handler. In
+         * simulation mode, just use a host function supplied in the ecall
+         * context. */
+        if (saved_fs)
```

Comment:
Since the goal is to restore FS, we could just check whether the current FS[0] value is different from the FS[0] value prior to the OCALL. There does not seem to be a strong reason to rely on GS[0].

Reply:
done